src/main/java/com/tudu/todoapp/services/interfaces/ToDoItemService.java

```diff
@@ -0,0 +1,65 @@
+package com.tudu.todoapp.services.interfaces;
+
+import com.tudu.todoapp.entities.TodoItem;
+import com.tudu.todoapp.entities.TodoList;
+import com.tudu.todoapp.exceptions.ResourceConflictException;
+import com.tudu.todoapp.exceptions.ResourceNotFoundException;
+
+import java.util.List;
+
+/**
+ * Business logic related to {@code ToDoItem} entities.
+ */
+public interface ToDoItemService {
+
+    /**
+     * Returns all to do items.
+     */
+    List<TodoItem> getAllTodoItems();
+
+    /**
+     * Returns the to do item with that id.
+     *
+     * @throws ResourceNotFoundException if the board was not found.
+     */
+    TodoItem getToDoItemById(Long itemId) throws ResourceNotFoundException;
+
+    /**
+     * Returns a page containing to do items.
+     *
+     * @throws ResourceNotFoundException if the page was not found.
+     */
+    List<TodoItem> getTodoItemsPage (int pageNumber, int perPage) throws ResourceNotFoundException;
```

Comment:
AAAAAAAAAAAA MAHNI INTERVALA SLED IMETO NA METODAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAa

Reply:
also this shouldnt throw ResourceNotFoundException, it just returns a List doesnt it. If there are no elements it just returns an empty list.

Reply:
Well, right, I was thinking about the situation when a page you are trying to reach does not exist, but I am retrospecting on it right now and it seems that this cannot happen. Fixing it.